src/algorithms/digi/SiliconPulseGeneration.h

```diff
@@ -33,7 +36,47 @@ using SiliconPulseGenerationAlgorithm =
     algorithms::Algorithm<algorithms::Input<edm4hep::SimTrackerHitCollection>,
                           algorithms::Output<PulseType::collection_type>>;
 
-class SignalPulse;
+class SignalPulse {
+
+public:
+  virtual ~SignalPulse()                                = default; // Virtual destructor
+  virtual double operator()(double time, double charge) = 0;
```

Comment:
This getter must be `const`. It should have always been `const`, and the `EvaluatorPulse` will need to deal with the map of parameters in a different way.